docs/tutorial/file_handling.md

```diff
@@ -0,0 +1,471 @@
+# File Handling
+
+A scripting library does not only contain reading and writing operations, that is why we also include methods for manipulating files such as creations, deletions, permissions, and other nice stuff in the library.
+
+## Create a file
+
+You may want to create a file without immediately writing on it. For example, when you want to modify the permissions first or let another process/fiber handle it instead. For such and more reasons, you can create a file or directory using the `createFile` and `createDirectory` functions.
+
+This is how you can create a file in Shellfish:
+
+@:select(api-style)
+
+@:choice(syntax)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.syntax.path.*
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object Creating extends IOApp.Simple:
+
+  val filePath = Path("path/to/your/desired/creation/NiceScript.scala")
+
+  def run: IO[Unit] =
+    for
+      _ <- filePath.createFile
+      created <- filePath.exists
+      _ <- IO.println(s"File created? $created")
+    yield ()
+
+end Creating
+```
+
+@:choice(static)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.FilesOs
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object Creating extends IOApp.Simple:
+
+  val filePath = Path("path/to/your/desired/creation/NiceScript.scala")
+
+  def run: IO[Unit] =
+    for
+      _ <- FilesOs.createFile(filePath)
+      created <- FilesOs.exists(filePath)
+      _ <- IO.println(s"File created? $created")
+    yield ()
+
+end Creating
+```
+
+@:@
+
+Here, we are first creating the file using the `createFile` method and then checking its existent with the `exists` method.
+
+**Important:** The `createFile` and `createDirectory` methods will only work if the parent directory already exists, otherwise it will fail so be careful with that. Because of that, if you want to create the parent directories, as well you should use the `createDirectories` method like this:
```

Comment:
```suggestion
**Important:** The `createFile` and `createDirectory` methods will only work if the parent directory already exists, and fail otherwise. The `createDirectories` method will recursively create the directories instead:
```